sdk/src/main/scala/coop/rchain/sdk/dag/data/DagData.scala

```diff
@@ -0,0 +1,27 @@
+package coop.rchain.sdk.dag.data
+
+/**
+  * Represents field accessors for message and sender.
+  *
+  * @tparam M represents a message
+  * @tparam MId represents a message identifier (message hash)
+  * @tparam S represents a sender
+  * @tparam SId represents a sender identifier (public key)
+  */
+trait DagData[M, MId, S, SId] {
+  /* Message */
+
+  def seqNum(m: M): Long
+
+  def blockNum(m: M): Long
+
+  def justifications(m: M): Set[MId]
+
+  def sender(m: M): SId
+
+  def bondsMap(m: M): Map[SId, Long]
+
+  /* Sender */
+
+  def stake(s: S): Long
```

Comment:
I'm thinking that this might be a bit misleading, as there is no notion of a stake for sender in general - sender has a stake only inside some state, and sender can have different stake in different states. 
And also you already have `def bondsMap(m: M): Map[SId, Long]`, which gives the stake of sender as per some message.

Reply:
Good point. I'll remove it.